src/sentry/statistical_detectors/algorithm.py

```diff
@@ -135,11 +146,69 @@ def update(self, payload: DetectorPayload) -> Optional[TrendType]:
 
         return TrendType.Unchanged
 
-    @property
-    def state(self) -> MovingAverageCrossOverDetectorState:
-        return MovingAverageCrossOverDetectorState(
-            timestamp=self.timestamp,
-            count=self.count,
-            moving_avg_short=self.moving_avg_short.value,
-            moving_avg_long=self.moving_avg_short.value,
-        )
+
+@dataclass(frozen=True)
+class MovingAverageRelativeChangeDetectorConfig(MovingAverageDetectorConfig):
+    threshold: float
+
+
+class MovingAverageRelativeChangeDetector(MovingAverageDetector):
+    def __init__(
+        self,
+        state: MovingAverageDetectorState,
+        config: MovingAverageRelativeChangeDetectorConfig,
+    ):
+        super().__init__(state, config)
+        self.threshold = abs(config.threshold)
+
+    def update(self, payload: DetectorPayload) -> Optional[TrendType]:
+        if self.timestamp is not None and self.timestamp > payload.timestamp:
+            # In the event that the timestamp is before the payload's timestamps,
+            # we do not want to process this payload.
+            #
+            # This should not happen other than in some error state.
+            logger.warning(
+                "Trend detection out of order. Processing %s, but last processed was %s",
+                payload.timestamp.isoformat(),
+                self.timestamp.isoformat(),
+            )
+            return None
+
+        old_moving_avg_short = self.moving_avg_short.value
+        old_moving_avg_long = self.moving_avg_long.value
+
+        self.moving_avg_short.update(payload.value)
+        self.moving_avg_long.update(payload.value)
+        self.timestamp = payload.timestamp
+        self.count += 1
+
+        # The heuristic isn't stable initially, so ensure we have a minimum
+        # number of data points before looking for a regression.
+        stablized = self.count > self.config.min_data_points
+
+        try:
+            relative_change_old = (old_moving_avg_short - old_moving_avg_long) / abs(
+                old_moving_avg_long
+            )
+            relative_change_new = (self.moving_avg_short.value - self.moving_avg_long.value) / abs(
+                self.moving_avg_long.value
```

Comment:
Do we handle a condition when a new value is 0? We should probably ignore 0 values and not use them to calculate moving averages

Reply:
Are you talking about when the p95 for the current time bucket is 0? If so, there 2 cases where this can happen
1. No data was received for the current time bucket. In which case, the update/check is skipped entirely
2. Data was received and they were 0. In which case, I think it makes sense to do the update because the SDK is reporting a duration of 0 which could indicate something has changed.

Reply:
yeah, I dont think scenario 2 is possible in practice but I think this is still the correct handling